src/rest.rs

```diff
@@ -627,6 +627,40 @@ fn handle_request(
         path.get(3),
         path.get(4),
     ) {
+        (&Method::GET, Some(&"headers"), hash, count, None, None) => {
+            let count = count
+                .and_then(|c| c.parse::<usize>().ok())
+                .and_then(|c| match c {
+                    0 => Some(1),
+                    1..=2000 => Some(c),
+                    _ => None,
+                })
+                .unwrap_or(2000);
+            let from_hash = hash
+                .map(|h| BlockHash::from_hex(h))
+                .transpose()?
+                .unwrap_or_else(|| query.chain().best_hash());
+
+            let headers = match query.chain().get_headers(&from_hash, count) {
+                Some(headers) => headers,
+                None => return Err(HttpError::not_found("Block not found".to_string())),
+            };
+
+            let mut raw = Vec::with_capacity(8 + 80 * headers.len());
```

Comment:
The header size is different in elements. I think this should be the only difference there.